File: sample/ts/pytra/utils/assertions.js
exports.py_assert_true = function(cond, _label) { return !!cond; };
exports.py_assert_eq = function(actual, expected, _label) { return actual === expected; };
exports.py_assert_all = function(results, _label) {
  if (!Array.isArray(results)) return false;
  for (const v of results) { if (!v) return false; }
  return true;
};
exports.py_assert_stdout = function(_expected_lines, fn) {
  if (typeof fn === 'function') { fn(); }
  return true;
};

File: src/pytra/utils/assertions.py
from __future__ import annotations

def py_assert_true(cond: bool, label: str = "") -> bool:
    if cond:
        return True
    if label != "":
        print("[assert_true] " + label + ": False")
    else:
        print("[assert_true] False")
    return False


def py_assert_eq(actual: int | str | bool | None, expected: int | str | bool | None, label: str = "") -> bool:
    ok: bool = str(actual) == str(expected)
    if ok:
        return True
    if label != "":
        print("[assert_eq] " + label + ": actual=" + str(actual) + ", expected=" + str(expected))
    else:
        print("[assert_eq] actual=" + str(actual) + ", expected=" + str(expected))
    return False


def py_assert_all(results: list[bool], label: str = "") -> bool:
    for v in results:
        if not v:
            if label != "":
                print("[assert_all] " + label + ": False")
            else:
                print("[assert_all] False")
            return False
    return True


def py_assert_stdout(expected_lines: list[str], fn: callable[[], None]) -> bool:
    # self_hosted parser / runtime 互換優先: stdout capture は未実装。
    return True
